punishments.js

```diff
@@ -843,6 +845,56 @@ Punishments.unnamelock = function (name) {
 	}
 	return success;
 };
+/**
+ * @param {User} user
+ * @param {number} expireTime
+ * @param {string} id
+ * @param {...string} [reason]
+ * @return {PunishmentRow}
+ */
+Punishments.battleban = function (user, expireTime, id, ...reason) {
+	if (!id) id = user.getLastId();
+
+	if (!expireTime) expireTime = Date.now() + BATTLEBAN_DURATION;
+	let punishment = ['BATTLEBANNED', id, expireTime, ...reason];
+
+	let affected = Punishments.punish(user, punishment);
+	for (let curUser of affected) {
+		curUser.battleBanned = id;
+		curUser.updateIdentity();
+	}
+
+	return affected;
+};
+/**
+ * @param {string} name
+ */
+Punishments.unbattleban = function (name) {
+	let user = Users(name);
+	let id = toId(name);
+	/** @type {string[]} */
+	let success = [];
+	let unpunished = Punishments.unpunish(name, 'BATTLEBAN');
```

Comment:
You use ``BATTLEBAN`` in some places and ``BATTLEBANNED`` in others - pick one to consistently use